multi_host_migration/tests/migration_multi_host_auto_converge.py

```diff
@@ -0,0 +1,498 @@
+                self.capabilitys.pop()
+                self.capabilitys_state.pop()
+
+        @error.context_aware
+        def start_stress(self, pre_action=""):
+            """
+            start stress test on src before migration
+
+            :param pre_action: run cmd before start stress
+            """
+
+            if set_auto_converge == "no":
+                self.install_stressapptest()
+            logging.info("Try to login guest before migration test.")
+            vm = self.env.get_vm(self.params["main_vm"])
+            self.session = vm.wait_for_login(timeout=self.login_timeout)
+            if pre_action.strip() != "":
+                logging.info("run cmd '%s' in guest" % pre_action)
+                self.session.cmd(pre_action)
+
+            error.context("Do stress test before migration.", logging.info)
+            self.session.cmd(self.bg_stress_test)
+
+            def check_running():
+                return self.session.cmd_status(self.check_running_cmd) == 0
+
+            if self.check_running_cmd:
+                if not utils_misc.wait_for(check_running, timeout=360):
+                    raise error.TestFail("Failed to start '%s' in guest." %
+                                         self.bg_stress_test)
+
+        @error.context_aware
+        def load_host(self):
+
+            """
+            retrieve or set a process's CPU affinity
+            """
+
+            error.context("load host before migration.", logging.info)
+            utils.run(self.load_host_cmd)
+
+        @error.context_aware
+        def get_sar_output(self):
+
+            """
+            get output of command sar during migration
+
+            :param vm: vm object
+            """
+
+            error.context("Get output of command sar during migration",
+                          logging.info)
+            vm = self.env.get_vm(self.params["main_vm"])
+            while vm.is_alive():
+                session = vm.wait_for_login(timeout=self.login_timeout)
+                s, o = session.cmd_status_output(get_sar_output_cmd)
+                if s != 0:
+                    raise error.TestFail("Failed to get sar output in guest."
+                                         "The detail is: %s" % o)
+                session.close()
+            o = o.split("\n")
+            for i in range(len(o)):
+                if sar_cpu_str in o[i]:
+                    cpu_average = o[i+1].split()[2:]
+                    cpu_average = map(float, cpu_average)
+                if sar_memory_str in o[i]:
+                    memory_average_raw = o[i+1].split()
+                    memory_average = []
+                    for j in range(len(memory_average_raw)):
+                        if j == 3 or j == 7:
+                            memory_average.append(memory_average_raw[j])
+                    memory_average = map(float, memory_average)
+                    break
+            logging.info("cpu average list: %s" % cpu_average)
```

Comment:
3495     File "/usr/local/autotest/tests/virt/test-providers.d/downloads/io-github-autotest-qemu/multi_host_migration/tests/migration_multi_host_auto_converge.py", line 175, in get_sar_output
3496       logging.info("cpu average list: %s" % cpu_average)                        
3497   UnboundLocalError: local variable 'cpu_average' referenced before assignment